test/Npgsql.Tests/ConnectionUriTests.cs

```diff
@@ -0,0 +1,696 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace Npgsql.Tests
+{
+    public class ConnectionUriTests : TestBase
+    {
+        [Test]
+        public void InvalidArgument()
+        {
+            Assert.That(() => new NpgsqlConnectionUri(null!),
+                Throws.InstanceOf(typeof(ArgumentNullException)));
+
+            Assert.That(() => new NpgsqlConnectionUri(""),
+                Throws.InstanceOf(typeof(UriFormatException)));
+        }
+
+        [Test]
+        public void InvalidConnectionUri()
```

Comment:
nit: if you want, InvalidArgument and InvalidConnectionURi can be a single parameterized test, where each TestCase gives the URI and the expected exception